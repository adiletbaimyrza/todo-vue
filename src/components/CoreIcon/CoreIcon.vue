<template>
  <img
    :class="$style['icon']"
    :src="src"
    :width="width"
    :height="height"
    alt="Icon"
  />
</template>

<script>
export default {
  name: 'CoreIcon',
  props: {
    src: {
      type: String,
      required: true,
    },
    width: {
      type: String,
      default: '24px',
    },
    height: {
      type: String,
      default: '24px',
    },
  },
}
</script>

<style lang="scss" module>
.icon {
  display: block;
}
</style>
