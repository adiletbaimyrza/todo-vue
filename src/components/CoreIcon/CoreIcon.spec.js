import { shallowMount } from '@vue/test-utils'
import CoreIcon from './CoreIcon.vue'

/* global describe it expect */
describe('CoreIcon.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(CoreIcon, {
      props: {
        src: 'test-icon.png',
      },
    })
    expect(wrapper.exists()).toBe(true)
  })

  it('renders with correct src, width, and height props', () => {
    const wrapper = shallowMount(CoreIcon, {
      props: {
        src: 'test-icon.png',
        width: '32px',
        height: '32px',
      },
    })
    const img = wrapper.find('img')
    expect(img.attributes('src')).toBe('test-icon.png')
    expect(img.attributes('width')).toBe('32px')
    expect(img.attributes('height')).toBe('32px')
  })

  it('applies default width and height when not provided', () => {
    const wrapper = shallowMount(CoreIcon, {
      props: {
        src: 'test-icon.png',
      },
    })
    const img = wrapper.find('img')
    expect(img.attributes('width')).toBe('24px')
    expect(img.attributes('height')).toBe('24px')
  })
})
