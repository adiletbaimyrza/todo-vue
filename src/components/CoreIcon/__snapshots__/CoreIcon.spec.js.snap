// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreIcon.vue renders correctly 1`] = `
<img
  alt="Icon"
  class="icon"
  height="24px"
  src="test-icon.png"
  width="24px"
/>
`;
